articles/house-price-prediction/index.md

```diff
@@ -0,0 +1,138 @@
+
+Once the model is fit, we evaluate the model’s performance on the test set we got earlier. We use the `predict` method present in the `RandomForestRegressor` class. The `predict` method takes in the test input data and predicts an output. Using the predicted output and the actual output known from the dataset, we compute the test accuracy.
+
+Another useful evaluation metric is the `Mean Squared Error`. The MSE loss gives an estimate of how far the prediction is from the mean of the output. Computing the MSE gives us an idea about the performance of the algorithm.
+
+### Code
+We'll use the dataset that is stored in a CSV file for a detailed view and easier access. Download the CSV file from this [link](https://drive.google.com/file/d/1clV931HTopTlD7ZWLotFSbsr9SAX50S8/view?usp=sharing). Save the CSV file in the same directory as the python file.
+
+```py
+# load the libraries
+import sklearn
+from sklearn.metrics import mean_squared_error as MSE
+from sklearn. pre-processing import LabelEncoder
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.datasets import load_boston
+
+import pandas as pd
+import numpy as np
+# load the data from load_boston function
+data = load_boston()
+# store the list of features in the dataset using feature_names
+array = data.feature_names
+array = np.append(array,['MEDV'])
+# separate the data and target values in the dataset
+data, target = data.data, data.target
+# split the data into training and test set to avoid overfitting
+Xtrain, Xtest, Ytrain, Ytest = train_test_split(data,target,test_size=0.3)
+
+print(Xtrain.shape,Ytrain.shape)
+
+TARGET_PRICE = 'MEDV'
+
+# instantiate the LabelEncoder
+le = LabelEncoder()
+# the data is stored in a csv file for your reference
+df = pd.read_csv('./boston.csv')
+
+y = df[TARGET_PRICE]
+# drop the target price column, as this is the training_data
+df = df.drop([TARGET_PRICE], axis=1)
+# tranform the categorical values into numerical values
+df['CHAS'] = le.fit_transform(df['CHAS'])
+
+x = df
+# instantiate the RandomForestRegressor with all the processors available
+dt = RandomForestRegressor(criterion='mae',n_jobs=-1, n_estimators=10,max_depth=6, min_samples_leaf=1, random_state=3)
+# fit the random forest to training data
+dt.fit(Xtrain,Ytrain)
+
+# predict the output for test data
+y_predicted = dt.predict(Xtest)
+# find the accuracy of predcition using training data
+accuracy = dt.score(Xtest,Ytest)
+# compute the Mean Square error using MSE function from sklearn.metrics module.
+MSE_score = MSE(Ytest,y_predicted)
+
+# print the final results
+print("Training Accuracy:",dt.score(Xtrain,Ytrain))
+print("Testing Accuracy:",accuracy)
+print("Mean Squared Error",MSE_score.mean())
+```
+
+The code output should look something like this:
+
+![output](/engineering-education/house-price-prediction/output.jpg)
```

Comment:
You could consider copy-pasting the terminal output and placing it in a code fence with bash as the language.